include/bsp/iotc_bsp_io_net.h

```diff
@@ -18,73 +18,88 @@
 #define __IOTC_BSP_IO_NET_H__
 
 /**
- * \mainpage Device SDK Board Support Package
- *
- * # Welcome
- * The Board Support Package (BSP) hosts the platform-specific code for the
- * Device SDK.
- *
- * Modify the BSP accordingly to port the Device SDK to new platforms. Other
- * functionality, such as the Device SDK source, event system, MQTT serializer,
- * and callback system, isn't part of the BSP.
- *
- * # Getting started
- * The Device SDK includes a POSIX implementation of the BSP. Use this
- * implementation for for Linux desktops and devices.
- *
- * To customize the BSP for non-POSIX platforms, see the <a
- * href="../../../porting_guide.md">porting guide</a>.
- * <code>iotc_bsp_rng.h</code>, <code>iotc_bsp_time.h</code>, and
- * <code>iotc_bsp_io_fs.h</code>.
- *
- * # BSP source files
- * The BSP consists of the following files. Each file meets a platform library
- * requirement.
- *  - <code>iotc_bsp_crypto.h</code> implements a cryptography library to sign
- * JWTs.
- *  - <code>iotc_bsp_io_net.h</code> implements asychronous networking.
- *  - <code>iotc_bsp_mem.h</code> allocates platform memory.
- *  - <code>iotc_bsp_rng.h</code> implements random number generation.
- *  - <code>iotc_bsp_time.h</code> implements time functions.
- *  - <code>iotc_bsp_tls.h</code> implements Transport Layer Security (TLS).
- *  - (Optional) <code>iotc_bsp_io_fs.h</code> manages the file system to load
- * certificates.
- *
- * # TLS BSPs
- * The Device SDK includes the following out-of-the-box TLS implementations.
- *
- * ## mbedTLS
- * The default <code>make</code> target <a href"~/src/bsp/mbedtls">downloads and
- * builds</a> <a href"https://tls.mbed.org">mbedTLS</a>.
- *
- * ## wolfSSL
- * See the <a href="../../../user_guide.md">user guide</a> and <a
- * href="../../../porting_guide.md">porting guide</a> to configure the Device
- * SDK for wolfSSL.
- *
- * # Further Reading
- * <ul><li>Device SDK <a href="../../api/html/index.html">API reference</a></li>
- * <li>Device SDK <a href="../../../user_guide.md">user guide</a></li>
- * <li>Device SDK <a href="../../../porting_guide.md">porting guide</a></li>
- * </ul>
+ * @mainpage Overview
```

Comment:
Ah, so this mainpage content is here. Because there is no better choice, right? Is has to exist somewhere, just asking to ensure.

Reply:
Ya, but we can move it too a new location if you'd prefer (not too much work). If one of these files is the "main" file then we can use that one... We can also create new file just for the mainpage content.